packages/x-components/src/x-modules/facets/entities/single-select.modifier.ts

```diff
@@ -0,0 +1,115 @@
+import {
+  isFacetFilter,
+  isHierarchicalFilter,
+  FacetFilter,
+  Facet,
+  Filter,
+  HierarchicalFilter
+} from '@empathyco/x-types';
+import { BaseFilterEntityModifier } from './types';
+
+/**
+ * Allows only to select only one filter from the same facet at the same time.
+ *
+ * @internal
+ */
+export class SingleSelectModifier extends BaseFilterEntityModifier {
+  /**
+   * Selects the passed filter, and then deselects any other filter of the same facet.
+   *
+   * @param filter - The filter to select.
+   */
+  select(filter: FacetFilter): void {
+    this.getOtherFilters(filter).forEach(this.deselect.bind(this));
+    this.entity.select(filter);
+  }
+
+  /**
+   * Retrieves the rest of selected filters of the same facet than the one given.
+   *
+   * @param filter - The filter to find its relatives.
+   * @returns A list of selected filters that belong to the same facet than the filter passed.
+   * @remarks It takes into account if the passed filter is an
+   * {@link @empathyco/x-types#HierarchicalFilter | HierarchicalFilter} to exclude the ancestors and
+   * descendants.
+   *
+   * @internal
+   */
+  protected getOtherFilters(filter: FacetFilter): FacetFilter[] {
+    /* This check seems dumb, but when you instantiate this modifier using the factory, the types
+      `FacetFilter` parameter type is lost, so we should check it to avoid unexpected crashes
+      due to a wrong configuration. */
+    if (isHierarchicalFilter(filter)) {
+      const ancestorsIds = this.getAncestorsIds(filter);
```

Comment:
Yes, the problem is when we save the response into the state, and call the entity to reset the state. Since the entity here is again the SingleSelectModifier, this deselects the rest of the filters, and depending on the order of the filters, we can get a wrong state with the HierarchicalFilters.